### PR TITLE
Make specularGlossiness inherit from MeshStandardMaterial

### LINKED ISSUE #18463: SpecularGlossiness materials no longer reflect the environment
##### Description of the problem

Something happened between r110 and dev, where glTF with SpecularGlossiness aren't playing well with PMREM anymore (other lights work).

To repro, put [this model](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/SpecGlossVsMetalRough) in Don's [viewer](https://gltf-viewer.donmccurdy.com/) and turn off the ambient and direct lights and you'll see this:
image

##### Three.js version

- [x] Dev
- [ ] r112
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by elalish**
Fixes #18463 

I've just taken #14099 by @pailhead and fixed the conflicts to update it. I made only a couple of minor fixes of my own that you can see in the last commit. This replaces my first failed effort at fixing this problem in #18465; the approach from this old PR both fixes the problem and also cleans up the code nicely. It also makes it unnecessary to delve into the gltf.parser.extensions to allow this material to be cloned; it just happens automatically!

### CODE REVIEW

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -603,6 +603,158 @@ THREE.GLTFLoader = ( function () {
...
+
+		/*eslint-disable*/
+		Object.defineProperties(
+			this,
+			{	
+				specular: {
+					get: function () { return uniforms.specular.value; },
+					set: function ( v ) { uniforms.specular.value = v; }
+        },
+        specularMap: {
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: Minor indentation issue.

**Comment by mrdoob** [comment]
Thanks!

**Comment by pailhead** [comment]
I think I need envoys for my PRs, when my name is directly behind them they tend to get stuck :)

I cleaned it up 6-7 times, but oh well, hopefully one of these days I’ll get some git blame.

**Comment by mrdoob** [comment]
Try submitting your next PR with a different username 😝

**Comment by pailhead** [comment]
Looks like the way to go :)

**Pull request merged**